tests/transforms/test_control.py

```diff
@@ -0,0 +1,198 @@
+from functools import partial
+import numpy as np
+import pennylane as qml
+from pennylane.tape import QuantumTape
+from pennylane.transforms.control import ctrl, ControlledOperation
+from pennylane.tape.tape import expand_tape
+
+
+def assert_equal_operations(ops1, ops2):
+    """Assert that two list of operations are equivalent"""
+    assert len(ops1) == len(ops2)
+    for op1, op2 in zip(ops1, ops2):
+        assert type(op1) == type(op2)
+        assert op1.wires == op2.wires
+        np.testing.assert_allclose(op1.parameters, op2.parameters)
+
+
+def test_control_sanity_check():
+    """Test that control works on a very standard usecase."""
+    def make_ops():
+        qml.RX(0.123, wires=0)
+        qml.RY(0.456, wires=2)
+        qml.RX(0.789, wires=0)
+        qml.PauliX(wires=2)
+        qml.PauliY(wires=4)
+        qml.PauliZ(wires=0)
+
+    with QuantumTape() as tape:
+        cmake_ops = ctrl(make_ops, control=1)
+        #Execute controlled version.
+        cmake_ops()
+
+    expected = [
+        qml.CRX(0.123, wires=[1, 0]),
+        qml.CRY(0.456, wires=[1, 2]),
+        qml.CRX(0.789, wires=[1, 0]),
+        qml.CNOT(wires=[1, 2]),
+        qml.CY(wires=[1, 4]),
+        qml.CZ(wires=[1, 0]),
+    ]
+    assert len(tape.operations) == 1
+    ctrl_op = tape.operations[0]
+    assert isinstance(ctrl_op, ControlledOperation)
+    assert len(ctrl_op.subtape.operations) == 6
+    expanded = ctrl_op.expand()
+    assert_equal_operations(expanded.operations, expected)
+
+
+def test_adjoint_of_control():
+    """Test adjoint(ctrl(fn)) and ctrl(adjoint(fn))"""

```

Comment:
the docstring really helps lol. See my previous comment, where I thought from the function name it was just adjoint of control 🤦